packages/flutter/test/widget/scroll_events_test.dart

```diff
@@ -136,4 +136,21 @@ void main() {
       expect(log, equals(['scrollstart', 'scrollend']));
     });
   });
+
+  test('fling up ends', () {
+    testWidgets((WidgetTester tester) {
+      GlobalKey<ScrollableState<Scrollable>> scrollKey = new GlobalKey<ScrollableState<Scrollable>>();
+      List<String> log = <String>[];
+      tester.pumpWidget(_buildScroller(key: scrollKey, log: log));
+
+      expect(log, equals([]));
+      tester.flingFrom(new Point(100.0, 100.0), new Offset(50.0, 50.0), 500.0);
+      tester.pump(new Duration(seconds: 1));
+      tester.pump(new Duration(seconds: 1));
+      tester.pump(new Duration(seconds: 1));
+      log.removeWhere((String value) => value == 'scroll');
+      expect(log, equals(['scrollstart', 'scrollend']));
```

Comment:
could also just check that the first is a scroll, the last is an end, and that nothing is left if you remove the scrolls.